### PR TITLE
Feature: Mineshaft click to gfs ascension rope

### PR DESCRIPTION
**Pull request by j10a1n15**
## What
This Pull Request adds a feature which allows to user to get an Ascension Rope from sack.
With this Item you can now leave the mineshaft and still get the leaving rewards.

<details>
<summary>Images</summary>

I changed the text to yellow, didnt bother getting a new screenshot
image

</details>

## Changelog New Features
+ Click to get an Ascension Rope from sacks in the Mineshaft. - j10a1n15

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/mining/MiningNotifications.kt [hunk 1]
```diff
@@ -61,6 +69,13 @@ object MiningNotifications {
             scrapDrop.matches(message) -> sendNotification(MiningNotificationList.SCRAP)
             goldenGoblinSpawn.matches(message) -> sendNotification(MiningNotificationList.GOLDEN_GOBLIN)
             diamondGoblinSpawn.matches(message) -> sendNotification(MiningNotificationList.DIAMOND_GOBLIN)
+            frostbitePattern.matches(message) -> {
+                if (IslandType.MINESHAFT.isInIsland() && config.getAscensionRope) {
+                    runDelayed(0.5.seconds) {
+                        ChatUtils.clickableChat("Click here to get an Ascension Rope!", "/gfs ASCENSION_ROPE 1")
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/mining/MiningNotifications.kt]: please use the gefFromSackApi. (Has a clickable chat variant)

**Comment by Thunderblade73** [comment]
This new function, I don't like. Since creating a primitivItem stack is simple to do and also is possible to cach in this case. The method btw. is .toPrimitivStack for NeuInternalNames (not sure if I mistyped the function name).

File: src/main/java/at/hannibal2/skyhanni/api/GetFromSackAPI.kt [hunk 2]
```diff
@@ -58,6 +58,13 @@ object GetFromSackAPI {
     ) =
         ChatUtils.clickableChat(text, "${commands.first()} ${item.internalName.asString()} ${item.amount}")
 
+    fun getFromChatMessageSackItem(
+        item: NEUInternalName,
+        amount: Int,
+        text: String = "§lCLICK HERE§r§e to grab §ax${amount} §9${item.asString()}§e from sacks!"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/api/GetFromSackAPI.kt]:
```suggestion
        text: String = "§lCLICK HERE§r§e to grab §ax${amount} §9${item.itemName}§e from sacks!"
```

File: src/main/java/at/hannibal2/skyhanni/config/features/mining/MiningNotificationsConfig.java [hunk 3]
```diff
@@ -43,4 +43,10 @@ public class MiningNotificationsConfig {
     @ConfigOption(name = "Cold Threshold", desc = "Change when the Cold notification gets triggered.")
     @ConfigEditorSlider(minValue = 1, maxValue = 100, minStep = 1)
     public Property<Integer> coldThreshold = Property.of(50);
+
+    @Expose
+    @ConfigOption(name = "Get Ascension Rope", desc = "Click on a chat message to get an Ascension Rope when you're at 90 Cold and in the §bMineshaft§7. " +
+        "§cOnly works if you have an Ascension Rope in your sacks.")
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/mining/MiningNotificationsConfig.java]:
```suggestion
    @ConfigOption(
        name = "Get Ascension Rope",
        desc = "Click on a chat message to get an Ascension Rope when you're at 90 Cold and in the §bMineshaft§7. " +
            "§cOnly works if you have an Ascension Rope in your sacks."
    )
```

File: src/main/java/at/hannibal2/skyhanni/features/mining/MiningNotifications.kt [hunk 4]
```diff
@@ -46,6 +53,10 @@ object MiningNotifications {
         "goblin.diamondspawn",
         "§6A §r§bDiamond Goblin §r§6has spawned!"
     )
+    private val frostbitePattern by patternGroup.pattern(
+        "cold.frostbite",
+        "§9§lBRRR! §r§bYou're freezing! All you can think about is getting out of here to a warm campfire..."
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/mining/MiningNotifications.kt]:
```suggestion
        "§9§lBRRR! §r§bYou're freezing! All you can think about is getting out of here to a warm campfire\\.\\.\\."
```

**Review by hannibal002 — CHANGES_REQUESTED** [review]
small changes

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**